website/www/site/content/en/documentation/runtime/environments.md

```diff
@@ -17,54 +17,175 @@ limitations under the License.
+apache/beam_python3.6_sdk          latest              ...                 1 min ago         ...
+apache/beam_python3.7_sdk          latest              ...                 1 min ago         ...
+apache/beam_python3.8_sdk          latest              ...                 1 min ago         ...
+apache/beam_go_sdk                 latest              ...                 1 min ago         ...
+```
+
+If you did not provide a custom repo/tag as additional parameters (see below), you can retag the image and [push](https://docs.docker.com/engine/reference/commandline/push/) the image using Docker to a remote repository.
+
+```
+export BEAM_SDK_VERSION="2.26.0"
+export IMAGE_NAME="myrepo/mybeamsdk"
+export TAG="${BEAM_SDK_VERSION}-custom"
+
+docker tag apache/beam_python3.6_sdk "${IMAGE_NAME}:${TAG}"
+docker push "${IMAGE_NAME}:${TAG}"
+```
+
+**NOTE**: After pushing a container image, verify the remote image ID and digest matches the local image ID and digest output from `docker_images`
+
+##### Additional build parameters
+
+The docker Gradle task defines a default image repository and [tag](https://docs.docker.com/engine/reference/commandline/tag/) is the SDK version defined at [gradle.properties](https://github.com/apache/beam/blob/master/gradle.properties). The default repository is the Docker Hub `apache` namespace, and the default tag is the [SDK version](https://github.com/apache/beam/blob/master/gradle.properties) defined at gradle.properties.
+
+You can specify a different repository or tag for built images by providing parameters to the build task. For example:
+
 ```
-2. Customize the [Dockerfile](https://github.com/apache/beam/blob/master/sdks/python/container/Dockerfile). If you're adding dependencies from [PyPI](https://pypi.org/), use [`base_image_requirements.txt`](https://github.com/apache/beam/blob/master/sdks/python/container/base_image_requirements.txt) instead.
-3. [Reimage](#building-container-images) the container.
+./gradlew :sdks:python:container:py36:docker -Pdocker-repository-root="example-repo" -Pdocker-tag="2.26.0-custom"
+```
+
+builds the Python 3.6 container and tags it as `example-repo/beam_python3.6_sdk:2.26.0-custom`.
+
+From Beam 2.21.0 and later, a `docker-pull-licenses` flag was introduced to add licenses/notices for third party dependencies to the docker images. For example:
+
+```
+./gradlew :sdks:java:container:java8:docker -Pdocker-pull-licenses
+```
+creates a Java 8 SDK image with appropriate licenses in `/opt/apache/beam/third_party_licenses/`.
+
+By default, no licenses/notices are added to the docker images.
 
-### Testing customized images
 
-To test a customized image locally, run a pipeline with PortableRunner and set the `--environment_config` flag to the image path:
+## Using container images in pipelines
+
+The common method for providing a container image requires using the PortableRunner and setting the `--environment_config` flag to a given image path.
```

Comment:
Starting with Beam 2.25.0, prefer using `--environment_options=docker_container_image=foo` instead of `--environment_config`, since `--environment_config` causes confusion when used with different (non-Docker) environments (BEAM-10671).

Reply:
Do you think we need to note this? Or should we just recommend `--environment_options`. If it was only supported post 2.25.0, we can maybe say "Older versions of the SDK (<2.25.0) may not support `--environment_options`. For these versions, use `--environment_config=$IMAGE_URL` instead?"

Reply:
Since 2.25.0 is still pretty recent, maybe we should keep this as it is for now. Perhaps we can add a TODO to update this later?

Reply:
added a TODO; I think it makes sense to update this all together, maybe when we move to GA for custom containers or port options over to DataflowRunner.